Remove templates GCGLSpan & GCGLSpanTuple from GraphicsTypesGL.h

djg opened the pull request:
#### 747785fe59b8487f543fd0dea6c55d1aa10353ef
<pre>
Remove templates GCGLSpan &amp; GCGLSpanTuple from GraphicsTypesGL.h
<a href="https://bugs.webkit.org/show_bug.cgi?id=250817">https://bugs.webkit.org/show_bug.cgi?id=250817</a>
rdar://problem/104407319

Reviewed by Jer Noble.

To improve compile time of WebCoreArgumentCoders.h move the templates GCGLSpan &amp;
GCGLSpanTuple into their own header to be included by implementations of
GraphicsContextGL. So GraphicsContextGL.h can compile with just forward
declarations, move query functions getFloat, getBoolean, et al, from
GraphicsContextGL.h to GraphicsContextGL.cpp

Canonical link: <a href="https://commits.webkit.org/259124@main">https://commits.webkit.org/259124@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e4cb47fffcac46d7283df9f88bc11ddb195a7d2f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/103860 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/12977 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/36801 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113088 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173389 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/107811 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14000 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/3870 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96109 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112195 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109633 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10792 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/93858 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38504 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92623 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25468 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80156 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6333 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/26850 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6498 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3390 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12486 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46379 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6285 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8268 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

jernoble commented on Source/WebCore/platform/graphics/GraphicsTypesGL.h:
```diff
@@ -25,12 +25,8 @@
 
-#include <JavaScriptCore/ArrayBufferView.h>
 #include <cstdint>
 #include <limits>
```
I think this can be moved to GCGLSpan.h as well.

jernoble reviewed (approved):
r=me with nit

jernoble commented on Source/WebCore/platform/graphics/GraphicsContextGL.h:
```diff
@@ -46,6 +46,14 @@
 #endif
 #endif
 
+inline constexpr size_t gcGLSpanDynamicExtent = std::numeric_limits<size_t>::max();
+
+template<typename T, size_t Extent = gcGLSpanDynamicExtent>
+class GCGLSpan;
+
+template<typename... Types>
+struct GCGLSpanTuple;
+
```
This is super interesting. Perhaps this forward declaration belongs in GraphicsTypesGL.h?

djg replied:
It was there and I removed it from `GraphicsTypesGL.h` to make that header as-light-as-possible. (Mainly prompted by your previous comment, "I think this can be moved to GCGLSpan.h as well.")

djg committed:
Remove templates GCGLSpan & GCGLSpanTuple from GraphicsTypesGL.h
https://bugs.webkit.org/show_bug.cgi?id=250817
rdar://problem/104407319

Reviewed by Jer Noble.

To improve compile time of WebCoreArgumentCoders.h move the templates GCGLSpan &
GCGLSpanTuple into their own header to be included by implementations of
GraphicsContextGL. So GraphicsContextGL.h can compile with just forward
declarations, move query functions getFloat, getBoolean, et al, from
GraphicsContextGL.h to GraphicsContextGL.cpp

Canonical link: https://commits.webkit.org/259124@main
